### PR TITLE
ES6: Convert discussion/activity-stream

### PR DESCRIPTION
**Pull request by gu-stav**
ES6.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/discussion/activity-stream.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    applyState(html: string, streamType: string): void {
+        // update display
+        const $el = bonzo(this.elem).empty();
+
+        this.setState('loading');
+
+        $.create(html).each(el => {
+            $el.html($(el).html()).attr({ class: el.className });
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/activity-stream.js]: [Reserved words as property names](http://kangax.github.io/compat-table/es5/#test-Object/array_literal_extensions_Reserved_words_as_property_names) doesn't work in <= IE8, even with ES5 shim.

Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/activity-stream.js]: Then again in IE8 none of the CSS loads and the entire website looks like horror 😱  Maybe not worth worrying about 😅

File: static/src/javascripts/projects/common/modules/discussion/activity-stream.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        this.options.streamType = streamType;
+    }
+
+    change(opts: Object): Promise<any> {
+        this.setOptions(opts);
+        // eslint-disable-next-line no-underscore-dangle
+        return this._fetch();
+    }
+
+    fetched(resp: Object) {
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/activity-stream.js]: missing a return type

**Comment by SiAdcock** [comment]
Is there any need for the `bonzo` dependency in this module? Looks like we can replace it with `lib/$` (which is already being imported anyway). Not a major thing.

**Review by SiAdcock — APPROVED** [review]
Minor comments, but beautiful overall 💅

**Comment by gustavpursche** [comment]
Reverting this because of a Safari problem on profile pages.

**Pull request merged**